pandas/core/generic.py

```diff
@@ -11601,7 +11601,13 @@ def cum_func(self, axis=None, skipna=True, *args, **kwargs):
         def block_accum_func(blk_values):
             values = blk_values.T if hasattr(blk_values, "T") else blk_values
 
-            result = nanops.na_accum_func(values, accum_func, skipna=skipna)
+            if is_extension_array_dtype(values.dtype) and values.dtype.kind not in [
```

Comment:
I would rather have the actual logic live here than splitting it between the array functions and here.